### PR TITLE
refactor and add classes under "Person", resolve conflict files

### PR DESCRIPTION
**Pull request by EdwardKSG**
refactor Address to Major (ie. course of study), since address is not very relevant to our project while the students' major is. Method name, comments, string content, xml file etc are modified accordingly.

add new class Year (ie. year of study). parsers, junit tests and some UI stuff are all changed accordingly.

resolve the conflict files encountered in the previous PR attempt

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -19,18 +20,20 @@
     public static final String COMMAND_WORD = "add";
     public static final String COMMAND_ALIAS = "a";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. "
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the major book. "
```
Comment (Livian1107) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]: is the major book the new name for our product?

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -152,15 +157,16 @@ public EditPersonDescriptor(EditPersonDescriptor toCopy) {
...
+            setYear(toCopy.year);
             setTags(toCopy.tags);
         }
 
         /**
          * Returns true if at least one field is edited.
          */
         public boolean isAnyFieldEdited() {
-            return CollectionUtil.isAnyNonNull(this.name, this.phone, this.email, this.address, this.tags);
+            return CollectionUtil.isAnyNonNull(this.name, this.phone, this.email, this.major, this.tags);
```
Comment (Livian1107) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: how about this.year?

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 3]
```diff
@@ -32,9 +34,10 @@
      */
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE,
+                        PREFIX_EMAIL, PREFIX_MAJOR, PREFIX_YEAR, PREFIX_TAG);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL)
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_MAJOR, PREFIX_PHONE, PREFIX_EMAIL)
```
Comment (Livian1107) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java]: is year a compulsory field to fill in?

File: src/main/java/seedu/address/model/person/Email.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ public Email(String email) {
     }
 
     /**
-     * Returns if a given string is a valid person email.
+     * Returns if a given string is a valid person address.
```
Comment (Livian1107) [inline src/main/java/seedu/address/model/person/Email.java]: address?

File: src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java [hunk 5]
```diff
@@ -4,8 +4,8 @@
 import static org.junit.Assert.assertTrue;
 import static seedu.address.logic.commands.CommandTestUtil.DESC_AMY;
 import static seedu.address.logic.commands.CommandTestUtil.DESC_BOB;
-import static seedu.address.logic.commands.CommandTestUtil.VALID_ADDRESS_BOB;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_EMAIL_BOB;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_MAJOR_BOB;
```
Comment (Livian1107) [inline src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java:8]: how about year

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 6]
```diff
@@ -48,99 +54,110 @@
...
+                + YEAR_DESC_AMY, new AddCommand(expectedPerson));
     }
 
     @Test
     public void parse_compulsoryFieldMissing_failure() {
         String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE);
 
         // missing name prefix
-        assertParseFailure(parser, VALID_NAME_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB,
+        assertParseFailure(parser, VALID_NAME_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + MAJOR_DESC_BOB,
```
Comment (Livian1107) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: year?

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 7]
```diff
@@ -84,36 +88,41 @@ public void parse_invalidValue_failure() {
         assertParseFailure(parser, "1" + INVALID_NAME_DESC, Name.MESSAGE_NAME_CONSTRAINTS); // invalid name
         assertParseFailure(parser, "1" + INVALID_PHONE_DESC, Phone.MESSAGE_PHONE_CONSTRAINTS); // invalid phone
         assertParseFailure(parser, "1" + INVALID_EMAIL_DESC, Email.MESSAGE_EMAIL_CONSTRAINTS); // invalid email
-        assertParseFailure(parser, "1" + INVALID_ADDRESS_DESC, Address.MESSAGE_ADDRESS_CONSTRAINTS); // invalid address
+        assertParseFailure(parser, "1" + INVALID_MAJOR_DESC, Major.MESSAGE_MAJOR_CONSTRAINTS); // invalid major
         assertParseFailure(parser, "1" + INVALID_TAG_DESC, Tag.MESSAGE_TAG_CONSTRAINTS); // invalid tag
```
Comment (Livian1107) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: check for year?

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 8]
```diff
@@ -84,36 +88,41 @@ public void parse_invalidValue_failure() {
...
+        assertParseFailure(parser, "1" + TAG_DESC_FRIEND + TAG_EMPTY + TAG_DESC_HUSBAND,
+                Tag.MESSAGE_TAG_CONSTRAINTS);
+        assertParseFailure(parser, "1" + TAG_EMPTY + TAG_DESC_FRIEND + TAG_DESC_HUSBAND,
+                Tag.MESSAGE_TAG_CONSTRAINTS);
 
         // multiple invalid values, but only the first invalid value is captured
-        assertParseFailure(parser, "1" + INVALID_NAME_DESC + INVALID_EMAIL_DESC + VALID_ADDRESS_AMY + VALID_PHONE_AMY,
-                Name.MESSAGE_NAME_CONSTRAINTS);
+        assertParseFailure(parser, "1" + INVALID_NAME_DESC + INVALID_EMAIL_DESC
+                 + VALID_MAJOR_AMY + VALID_PHONE_AMY, Name.MESSAGE_NAME_CONSTRAINTS);
```
Comment (Livian1107) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: year?

**Review by Livian1107 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 3 above)
Comment (EdwardKSG) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java]: yes, but the content can be empty string

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 1 above)
Comment (EdwardKSG) [reply src/main/java/seedu/address/logic/commands/AddCommand.java]: no. it's a mistake

File: src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java (hunk 5 above)
Comment (EdwardKSG) [reply src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java:8]: cos didn't disign relevant tests

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 6 above)
Comment (EdwardKSG) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: yep this test should be updated

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 7 above)
Comment (EdwardKSG) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: no check for it. not all corresponding tests have been created

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 8 above)
Comment (EdwardKSG) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: no check of year is involved here. not all corresponding tests have been created

**Review by Livian1107 — APPROVED** [review]

**Pull request merged**